tests/rptest/services/native_kafka_consumer.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Vectorized, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import time
+import uuid
+
+from ducktape.services.background_thread import BackgroundThreadService
+from kafka import KafkaConsumer
+from rptest.wasm.topics_result_set import TopicsResultSet
+
+
+class NativeKafkaConsumer(BackgroundThreadService):
```

Comment:
This service doesn't really behave the way one would expect because the consumer itself is running on the coordination node (and also consumes a full node reservation without actually using it). I would recommend either (1) using normal python threading to run your producer/consumer and move that into the warm folder with a notice that the consumer/producer are running on the coordination node, or (2) expanding the capabilities of the existing consumer/producer services (that use `kaf` but that can be changed) that run the producer/consumer on the service's node reservation. This mean you probably can't use kafka-python because all the action has to happen over ssh. wrapping kafkacat, kaf, upstream kafka producer/consumer, etc... will be the best bet.

one reason to not run producer/consumer on coordination node is that in the cluster (non-docker) setup the coordination node may run on laptop while the cluster is in aws.

Reply:
Actually I don't think the service is running on the coordination node. The test fails unless I run with 6 nodes. I assume this is 3 for redpanda (because my test asks for 3) then an additional 3 because i an spawning 1 producer and 2 consumers.

Reply:
After a discussion its noted that although extra cluster resources are requested by ducktape, the actual source code run within these service is being executed within the orchestration container. The right way to work with the ducktape background service is to use the node object to communicate with the container over ssh